### PR TITLE
Unify model generation between DPG and HLC

### PR DESCRIPTION
**Pull request by AlexanderSher**
There are 11 new usages `Configuration.Generation1ConvenienceClient`, which will be either eliminated or replaced by a separate flag in separate PRs

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs [hunk 1]
```diff
@@ -103,6 +107,7 @@ public MgmtOutputLibrary()
             // these dictionaries are initialized right now and they would not change later
             RawRequestPathToOperationSets = CategorizeOperationGroups();
             ResourceDataSchemaNameToOperationSets = DecorateOperationSets();
+            _schemaToInputEnumMap = new CodeModelConverter(MgmtContext.CodeModel, MgmtContext.Context.SchemaUsageProvider).CreateEnums();
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:110]: is this temporary?  If so can you link the follow up item as a comment?

File: test/TestProjects/FirstTest-TypeSpec/tests/Generated/Samples/Samples_FirstTestTypeSpecClient.cs [hunk 2]
```diff
@@ -1350,7 +1350,7 @@ public async Task Example_FirstTestTypeSpec_DateTimeBody_ShortVersion_Async()
             Uri endpoint = new Uri("<https://my-service.azure.com>");
             FirstTestTypeSpecClient client = new FirstTestTypeSpecClient(endpoint);
 
-            using RequestContent content = RequestContent.Create("2022-05-10T14:57:31.2311892-04:00");
+            using RequestContent content = RequestContent.Create("2022-05-10T18:57:31.2311892Z");
```
Comment (m-nash) [inline test/TestProjects/FirstTest-TypeSpec/tests/Generated/Samples/Samples_FirstTestTypeSpecClient.cs:1353]: is the format change here intended?

File: test/TestProjects/ModelShapes/Generated/Models/InputModel.Serialization.cs [hunk 3]
```diff
@@ -623,7 +623,7 @@ internal static InputModel DeserializeInputModel(JsonElement element, ModelReade
                         array[index] = item.GetSingle();
                         index++;
                     }
-                    vectorReadOnlyRequiredNullable = new ReadOnlyMemory<float>?(array);
+                    vectorReadOnlyRequiredNullable = new ReadOnlyMemory<float>(array);
```
Comment (m-nash) [inline test/TestProjects/ModelShapes/Generated/Models/InputModel.Serialization.cs:626]: we lost the nullability here

File: test/TestProjects/TypeSchemaMapping/SomeFolder/Generated/Models/RenamedModelStruct.cs [hunk 4]
```diff
@@ -52,12 +51,8 @@ namespace CustomNamespace
         /// <param name="propertyToField"> . </param>
         /// <param name="fruit"> Fruit. </param>
         /// <param name="daysOfWeek"> Day of week. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="customizedFlattenedStringProperty"/> or <paramref name="propertyToField"/> is null. </exception>
         public RenamedModelStruct(string customizedFlattenedStringProperty, string propertyToField, CustomFruitEnum? fruit, CustomDaysOfWeek? daysOfWeek)
         {
-            Argument.AssertNotNull(customizedFlattenedStringProperty, nameof(customizedFlattenedStringProperty));
```
Comment (m-nash) [inline test/TestProjects/TypeSchemaMapping/SomeFolder/Generated/Models/RenamedModelStruct.cs:58]: lost assertions?

Comment (AlexanderSher) [reply test/TestProjects/TypeSchemaMapping/SomeFolder/Generated/Models/RenamedModelStruct.cs:58]: This is due to the difference in validation handling. In DPG, we don't validate optional fields, and if I recall correctly, we agreed to remove this validation from HLC, and from what I can tell, there are no such cases in azure-sdk-for-net - only in tests.

File: test/TestProjects/FirstTest-TypeSpec/tests/Generated/Samples/Samples_FirstTestTypeSpecClient.cs (hunk 2 above)
Comment (AlexanderSher) [reply test/TestProjects/FirstTest-TypeSpec/tests/Generated/Samples/Samples_FirstTestTypeSpecClient.cs:1353]: Yes. Default format for JSON body is RFC3339, default format for header is RFC1123
I have moved it into separate PR: https://github.com/Azure/autorest.csharp/pull/4371

File: test/TestProjects/ModelShapes/Generated/Models/InputModel.Serialization.cs (hunk 3 above)
Comment (AlexanderSher) [reply test/TestProjects/ModelShapes/Generated/Models/InputModel.Serialization.cs]: Nice catch, fixed

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:110]: Yes, should go away with this one: https://github.com/Azure/autorest.csharp/issues/3770

**Review by m-nash — APPROVED** [review]

**Pull request merged**